x-pack/plugins/threat_intelligence/public/modules/indicators/components/indicators_table/indicators_table.tsx

```diff
@@ -149,6 +152,20 @@ export const IndicatorsTable: VFC<IndicatorsTableProps> = ({
     return <EmptyState />;
   }
 
+  columns.map(
+    (col: EuiDataGridColumn) =>
+      (col.cellActions = [
+        ({ rowIndex, columnId, Component }: EuiDataGridColumnCellActionProps) => (
+          <AddToTimeline
+            data={indicators[rowIndex]}
+            field={columnId}
+            component={Component}
+            testId={CELL_TIMELINE_BUTTON_TEST_ID}
+          />
+        ),
+      ])
+  );
+
```

Comment:
I would do this mapping like this:

```typescript
const columnsWithCellActions = useMemo(
    () => columns.map(
    (col: EuiDataGridColumn) => ({
      ...col,
      cellActions: [
              ({ rowIndex, columnId, Component }: EuiDataGridColumnCellActionProps) => (
                <AddToTimeline
                  data={indicators[rowIndex]}
                  field={columnId}
                  component={Component}
                  testId={CELL_TIMELINE_BUTTON_TEST_ID}
                />
              ),
            ]
        }),
    ),
    []
);
```

this avoids data mutation and caches the  mapped columns so that there is not reference change when passed down to consumers via props.

Reply:
yup I need to focus more on these performance improvements, done!